be/src/exprs/variant_path_parser.cpp

```diff
@@ -254,25 +254,17 @@ StatusOr<VariantRowValue> VariantPath::seek(const VariantRowValue* variant, cons
     VariantValue current{variant->get_value().raw()};
     for (size_t seg_idx = 0; seg_idx < variant_path->segments.size(); ++seg_idx) {
         const auto& segment = variant_path->segments[seg_idx];
-
-        StatusOr<VariantValue> sub;
```

Comment:
reduce ctor of StatusOr<VariantValue>, which has memory allocation operations inside.